### PR TITLE
Improved playerlist.

### LINKED ISSUE #232: Selected player distinction
**Is your feature request related to a problem? Please describe.**
When you select a player, the only thing that changes is the name atop the player window.

**Describe the solution you'd like**
The button element of the selected player should be given a distinct color differentiating it from the others.

**Provide Reasoning**
It would greatly improve the user friendliness.

**Additional context**

image

### PR DESCRIPTION
**Pull request by aa15032261**
The highlighted player is now referred to the selected player instead of the host.
Added host, client, in vehicle and walking icons.

However, the in vehicle state is not very reliable as get_current_vehicle() doesnt return correct result sometimes.

Screenshot 2022-07-04 230218

### CODE REVIEW

**Review by LiamD-Flop — APPROVED** [review]

**Review by LiamD-Flop — CHANGES_REQUESTED** [review]
Are the .sfd and .ttf files necessary. If not please remove them

**Review by LiamD-Flop — DISMISSED** [review]
Is it possible for you to add a 3th icon for friends. So we have C, H and F

**Comment by aa15032261** [comment]
> Are the .sfd and .ttf files necessary. If not please remove them


The stf file can be opened with FontForge, which is the program I used to make the icon ttf, so I think we should keep the file in case someone else wants to add more icons into the program.

**Comment by aa15032261** [comment]
> Is it possible for you to add a 3th icon for friends. So we have C, H and F

Sure I can put a friend icon in

File: BigBaseV2/src/fonts/icons.sfd [hunk 1]
```diff
@@ -0,0 +1,309 @@
+SplineFontDB: 3.2
+FontName: Untitled1
+FullName: Untitled1
+FamilyName: Untitled1
+Weight: Regular
+Copyright: Copyright (c) 2022, Tom Chan
```
Comment (Yimura) [inline BigBaseV2/src/fonts/icons.sfd]: @GeopJr this seems like license problem to me, what do you think?

**Review by Yimura — CHANGES_REQUESTED** [review]
I have some questions about the font, from the **.sdf** file I can see you used a tool called FontForge.

Which icons did you use? How are the licensed and can we use them?
I've found some icon font generators online and this seems easier to use than having to manually edit a font.

File: BigBaseV2/src/fonts/icons.sfd (hunk 1 above)
Comment (GeopJr) [reply BigBaseV2/src/fonts/icons.sfd]: I'll reply in review since it involves more than that file.

**Review by GeopJr — DISMISSED** [review]
**IANAL**

For starters, the copyright notice seems wrong. These icons look like they belong to [Google's Material Symbols](https://fonts.google.com/icons).

That icon set is [licensed under Apache 2.0](https://github.com/google/material-design-icons/blob/master/LICENSE), which is [**NOT**](https://www.gnu.org/licenses/license-list.en.html#apache2) compatible with GPL-2.0:

> Please note that this license is not compatible with GPL version 2, because it has some requirements that are not in that GPL version. These include certain patent termination and indemnification provisions. The patent termination provision is a good thing, which is why we recommend the Apache 2.0 license for substantial programs over other lax permissive licenses.

I see at least two available options:

- YimMenu relicensing to a permissive license (we've discussed this on another platform).
- Using a different icon set, preferably under a Creative Commons license (see: https://github.com/iconify/icon-sets/blob/master/collections.md)

([CC BY 4.0](https://creativecommons.org/licenses/by/4.0/) requires only attribution)

**Comment by aa15032261** [comment]
> I have some questions about the font, from the **.sdf** file I can see you used a tool called FontForge.
> 
> Which icons did you use? How are the licensed and can we use them? I've found some icon font generators online and this seems easier to use than having to manually edit a font.



> **IANAL**
> 
> For starters, the copyright notice seems wrong. These icons look like they belong to [Google's Material Symbols](https://fonts.google.com/icons).
> 
> That icon set is [licensed under Apache 2.0](https://github.com/google/material-design-icons/blob/master/LICENSE), which is [**NOT**](https://www.gnu.org/licenses/license-list.en.html#apache2) compatible with GPL-2.0:
> 
> > Please note that this license is not compatible with GPL version 2, because it has some requirements that are not in that GPL version. These include certain patent termination and indemnification provisions. The patent termination provision is a good thing, which is why we recommend the Apache 2.0 license for substantial programs over other lax permissive licenses.
> 
> I see at least two available options:
> 
> * YimMenu relicensing to a permissive license (we've discussed this on another platform).
> * Using a different icon set, preferably under a Creative Commons license (see: https://github.com/iconify/icon-sets/blob/master/collections.md)
> 
> ([CC BY 4.0](https://creativecommons.org/licenses/by/4.0/) requires only attribution)

Sorry guys, I didn't pay attention to the licensing issues. So I have created some icons myself to replace the material icons used previously. Also, I switched to Fontello(https://github.com/fontello/fontello) to generate icon font.

The new playerlist menu looks like this:
Screenshot 2022-07-04 230218

File: BigBaseV2/src/views/players/view_players.cpp [hunk 2]
```diff
@@ -2,44 +2,97 @@
...
+			isHost = plyr->is_host();
+			isFriend = plyr->is_friend();
+			isInVehicle = plyr->get_current_vehicle() != nullptr;
+
+			/*Entity plyrPed = PLAYER::GET_PLAYER_PED_SCRIPT_INDEX(plyr->id());
+			if (ENTITY::DOES_ENTITY_EXIST(plyrPed)) {
+				if (PED::IS_PED_IN_ANY_VEHICLE(plyrPed, true)) {
+					isInVehicle = true;
+				}
+			}*/
```
Comment (Yimura) [inline BigBaseV2/src/views/players/view_players.cpp]: Remove unused code

**Review by Yimura — CHANGES_REQUESTED** [review]
Thanks for adapting to our requests!

I wanted to ask, did you test this on different resolutions?

**Comment by LiamD-Flop** [comment]
> Thanks for adapting to our requests!
> 
> I wanted to ask, did you test this on different resolutions?

I checked the changes. He used the same sizings as we did. So it should stay the same relative to our menu.

File: BigBaseV2/src/views/players/view_players.cpp (hunk 2 above)
Comment (aa15032261) [reply BigBaseV2/src/views/players/view_players.cpp]: done

**Review by Yimura — APPROVED** [review]

**Review by LiamD-Flop — APPROVED** [review]

**Pull request merged**